tests/functionnal/SLM.php

```diff
@@ -270,6 +270,229 @@ public function testLifecyle() {
+      $this->checkInput($scrit, $scrit_id, $scrit_in);
+      $this->checkInput($ocrit, $ocrit_id, $ocrit_in);
+      $this->checkInput($saction, $saction_id, $saction_in);
+      $this->checkInput($oaction, $oaction_id, $oaction_in);
+
+      // ## 2 - test using sla in tickets
+
+      // add rules for using sla
+      $ruleticket = new \RuleTicket;
+      $rulecrit   = new \RuleCriteria;
+      $ruleaction = new \RuleAction;
+
+      $ruletid = $ruleticket->add($ruleinput = [
+         'name'         => $this->method,
+         'match'        => 'AND',
+         'is_active'    => 1,
+         'sub_type'     => 'RuleTicket',
+         'condition'    => \RuleTicket::ONADD + \RuleTicket::ONUPDATE,
+         'is_recursive' => 1
+      ]);
+      $this->checkInput($ruleticket, $ruletid, $ruleinput);
+      $crit_id = $rulecrit->add($crit_input = [
+         'rules_id'  => $ruletid,
+         'criteria'  => 'name',
+         'condition' => 2,
+         'pattern'   => $this->method
+      ]);
+      $this->checkInput($rulecrit, $crit_id, $crit_input);
+      $act_id = $ruleaction->add($act_input = [
+         'rules_id'    => $ruletid,
+         'action_type' => 'assign',
+         'field'       => 'slas_id_tto',
+         'value'       => $sla1_id
+      ]);
+      $act_id = $ruleaction->add($act_input = [
+         'rules_id'    => $ruletid,
+         'action_type' => 'assign',
+         'field'       => 'slas_id_ttr',
+         'value'       => $sla2_id
+      ]);
+      $act_id = $ruleaction->add($act_input = [
+         'rules_id'    => $ruletid,
+         'action_type' => 'assign',
+         'field'       => 'olas_id_tto',
+         'value'       => $ola1_id
+      ]);
+      $act_id = $ruleaction->add($act_input = [
+         'rules_id'    => $ruletid,
+         'action_type' => 'assign',
+         'field'       => 'olas_id_ttr',
+         'value'       => $ola2_id
+      ]);
+      $this->checkInput($ruleaction, $act_id, $act_input);
+
+      // test create ticket
+      $ticket = new \Ticket;
+      $start_date = date("Y-m-d H:i:s", time() - 4 * MONTH_TIMESTAMP);
+      $tickets_id = $ticket->add($ticket_input = [
+         'date'    => $start_date,
+         'name'    => $this->method,
+         'content' => $this->method
+      ]);
+      $this->checkInput($ticket, $tickets_id, $ticket_input);
+      $this->integer((int)$ticket->getField('slas_id_tto'))->isEqualTo($sla1_id);
+      $this->integer((int)$ticket->getField('slas_id_ttr'))->isEqualTo($sla2_id);
+      $this->integer((int)$ticket->getField('olas_id_tto'))->isEqualTo($ola1_id);
+      $this->integer((int)$ticket->getField('olas_id_ttr'))->isEqualTo($ola2_id);
+      $this->string($ticket->getField('time_to_resolve'))->length->isEqualTo(19);
+
+      // test update ticket
+      $ticket = new \Ticket;
+      $tickets_id_2 = $ticket->add($ticket_input_2 = [
+         'name'    => "to be updated",
+         'content' => $this->method
+      ]);
+      $ticket->update([
```

Comment:
You should check that OLA/SLA fields are not already set before update.

Reply:
Fixed